jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/DefaultCredentialRetrievers.java

```diff
@@ -166,10 +166,23 @@ public DefaultCredentialRetrievers setCredentialHelper(@Nullable String credenti
       credentialRetrievers.add(inferredCredentialRetriever);
     }
 
+    String dockerConfigEnv = environment.get("DOCKER_CONFIG");
+    if (dockerConfigEnv != null) {
+      Path dockerConfigsPath = Paths.get(dockerConfigEnv);
+      credentialRetrievers.add(
+          credentialRetrieverFactory.dockerConfig(dockerConfigsPath.resolve(DOCKER_CONFIG_FILE)));
+      credentialRetrievers.add(
+          credentialRetrieverFactory.dockerConfig(
+              dockerConfigsPath.resolve(KUBERNETES_DOCKER_CONFIG_FILE)));
+      credentialRetrievers.add(
+          credentialRetrieverFactory.legacyDockerConfig(
+              dockerConfigsPath.resolve(LEGACY_DOCKER_CONFIG_FILE)));
```

Comment:
I could pull some of this out into a helper method probably, but I think we do want to add all 9 paths to the credential retriever list, since we want to fall back to as many options as possible if retrieving the credentials fails on one source.

Reply:
oh yeah you're right, I didn't realize we could add them all.